### PR TITLE
Modified things to things teach section, added supplement section and questions in check for understanding

### PR DESCRIPTION
**Pull request by yati1998**
Fixes issue #824

### CODE REVIEW

File: databases/relational-databases.md [hunk 1]
```diff
@@ -123,7 +79,7 @@ Because you must define a schema to store data in a relational database, some de
 
 ### Guided Practice
 
-- Install Postgres locally
+- [Install Postgres](/databases/installing-postgresql.md) locally
```
Comment (alodahl) [inline databases/relational-databases.md]: ❤️

File: databases/relational-databases.md [hunk 2]
```diff
@@ -32,67 +33,22 @@ To understand, query, and insert information into a relational database, a techn
 ### Specific Things To Teach
 
 - The specific problems a database solves
-  - Persistence of information
```
Comment (alodahl) [inline databases/relational-databases.md]: This is good info; it just doesn't belong in 'things to teach."  Can move the info into ### Lesson, please?

File: databases/relational-databases.md [hunk 3]
```diff
@@ -32,67 +33,22 @@ To understand, query, and insert information into a relational database, a techn
...
-    - without modifiers (entire table)
-    - with modifier constraints (same as `SELECT`)
-- DDL to create tables
-  - `CREATE TABLE`
 - Bonus topic for the advanced: foreign key constraints
 - Explain different databases with slightly different dialects (MySQL - PostgreSQL)
 - Explain existence of other types of non-relational, non-sql databases that will be encountered
-  - examples key-value (redis), graph (Neo4j), streaming (cassandra), document (mongo)
 
 ### Materials
```
Comment (alodahl) [inline databases/relational-databases.md]: Tell them what to do with each Materials link under ### Lesson.

File: databases/relational-databases.md [hunk 4]
```diff
@@ -167,6 +123,20 @@ Further learning: Object-Relational Mapping (& ORM libaries)
 - [JavaTPoint sql quiz](https://www.javatpoint.com/sql-quiz)
```
Comment (alodahl) [inline databases/relational-databases.md]: Needs new ‘check for understanding ‘ with questions instead (see #700 ). also, check off this outline in 700 while you’re there, please.

You can move these quizzes to ### Challenge or ### Supplemental Resources.

File: databases/relational-databases.md [hunk 5]
```diff
@@ -1,11 +1,12 @@
 # Relational Databases & Postgres
 
-### Projected Time
+### Projected Time: 5-6 hours
```
Comment (alodahl) [inline databases/relational-databases.md]: 😁

File: databases/relational-databases.md (hunk 4 above)
Comment (alodahl) [reply databases/relational-databases.md]: Base your questions on the ### Objectives.

File: databases/relational-databases.md (hunk 2 above)
Comment (yati1998) [reply databases/relational-databases.md]: so do I need to remove things to teach if I add it into lessons?

Comment (alodahl) [reply databases/relational-databases.md]: No

Comment (alodahl) [reply databases/relational-databases.md]: Look at the template to see how each section is used

Comment (alodahl) [reply databases/relational-databases.md]: Things to teach is perfect, but the things you removed should now go in lesson.

File: databases/relational-databases.md (hunk 4 above)
Comment (yati1998) [reply databases/relational-databases.md]: Sorry, but @alodahl  can you please explain what exactly I need to check in #700.

Comment (alodahl) [reply databases/relational-databases.md]: I just meant you should follow the rules in #700 .

File: databases/relational-databases.md [hunk 6]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
 - [SQL Fiddle](http://sqlfiddle.com)
 
 ### Lesson
+
```
Comment (alodahl) [inline databases/relational-databases.md]: Please move new material after the rest of the text that was already in ### lesson.

File: databases/relational-databases.md [hunk 7]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+  - `UPDATE...SET`
+    - without modifiers (entire table)
+    - with modifier constraints (same as `SELECT`)
+  - `DELETE FROM`
+    - without modifiers (entire table)
+    - with modifier constraints (same as `SELECT`)
+- DDL to create tables
+  - `CREATE TABLE`
+- Bonus topic for the advanced: foreign key constraints
+- Explain different databases with slightly different dialects (MySQL - PostgreSQL)
```
Comment (alodahl) [inline databases/relational-databases.md]: Do these 2 lines beginning with ‘explain ‘ ever get explained in this outline? If yes, these need to be rephrased. If no, please provide more information.

File: databases/relational-databases.md [hunk 8]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+- Explain existence of other types of non-relational, non-sql databases that will be encountered
+  - examples key-value (redis), graph (Neo4j), streaming (cassandra), document (mongo)
+
 - [Slides](https://drive.google.com/open?id=1xK7_t_yJcu4RcBkj0Gv-t5uyBCNr0g4cHKqAJSxNwY0) | [Video Walkthrough of Slides](https://drive.google.com/file/d/1V0bk3fH_8PsRE3Vz4J3qe3TTiqBClT6y/view)
 
 Also briefly review: [LucidChart: What is a Database Model?](https://www.lucidchart.com/pages/database-diagram/database-models). Be sure to read the [Relational Model](https://www.lucidchart.com/pages/database-diagram/database-models?a=1) section, and briefly skim over the rest just so you're aware that there are many types of database models. The only one we're working with here is the relational model, so don't worry about learning the rest yet. 
 
+- Go through the complete tutorial mentioned in the Materials Section. [SQL Fiddle](http://sqlfiddle.com) is the tool where you can implement your sql knowledge. Practically implement creating a database schema and peforming operation on it using the tool.
```
Comment (alodahl) [inline databases/relational-databases.md]: and perform operations on it using the  tool.

File: databases/relational-databases.md [hunk 9]
```diff
@@ -117,12 +166,25 @@ Schema design:
 Further learning: Object-Relational Mapping (& ORM libaries)
 - [freeCodeCamp: Which JavaScript ORM should you be using in 2018?](https://medium.freecodecamp.org/a-comparison-of-the-top-orms-for-2018-19c4feeaa5f)
 
-### Check for Understanding
-
+Solve the quizes mentioned below
```
Comment (alodahl) [inline databases/relational-databases.md]: quizzes

File: databases/relational-databases.md [hunk 10]
```diff
@@ -117,12 +166,25 @@ Schema design:
...
+
+### Check for Understanding
+
+- What do you mean by Relational database?
+- What is CRUD in sql?
+- What is DDL?
+- How does SELECT command works?
+- How does JOIN works?
+- What are the advantages of a Relational database?
+- What is ORMs?
```
Comment (alodahl) [inline databases/relational-databases.md]: What are ORMs,  Or  What is an ORM

File: databases/relational-databases.md [hunk 11]
```diff
@@ -117,12 +166,25 @@ Schema design:
...
 - [TutorialsPoint quiz](https://www.tutorialspoint.com/sql/sql_online_quiz.htm)
 
+
+### Check for Understanding
+
+- What do you mean by Relational database?
+- What is CRUD in sql?
+- What is DDL?
+- How does SELECT command works?
+- How does JOIN works?
```
Comment (alodahl) [inline databases/relational-databases.md]: Works -> work

File: databases/relational-databases.md [hunk 12]
```diff
@@ -117,12 +166,25 @@ Schema design:
...
 - [JavaTPoint sql quiz](https://www.javatpoint.com/sql-quiz)
 - [TutorialsPoint quiz](https://www.tutorialspoint.com/sql/sql_online_quiz.htm)
 
+
+### Check for Understanding
+
+- What do you mean by Relational database?
+- What is CRUD in sql?
+- What is DDL?
+- How does SELECT command works?
```
Comment (alodahl) [inline databases/relational-databases.md]: How does the SELECT command work?

File: databases/relational-databases.md [hunk 13]
```diff
@@ -117,12 +166,25 @@ Schema design:
...
+Solve the quizes mentioned below
 - [W3Schools quiz](http://www.w3schools.com/sql/sql_quiz.asp)
 - [JavaTPoint sql quiz](https://www.javatpoint.com/sql-quiz)
 - [TutorialsPoint quiz](https://www.tutorialspoint.com/sql/sql_online_quiz.htm)
 
+
+### Check for Understanding
+
+- What do you mean by Relational database?
+- What is CRUD in sql?
```
Comment (alodahl) [inline databases/relational-databases.md]: @Kundan28 @praveenscience is ‘sql’ correct, or is it always SQL?

File: databases/relational-databases.md [hunk 14]
```diff
@@ -117,12 +166,25 @@ Schema design:
...
+- What is DDL?
+- How does SELECT command works?
+- How does JOIN works?
+- What are the advantages of a Relational database?
+- What is ORMs?
+- How is PostgreSQL different from MySQL?
+- How to model data for relational databases?
```
Comment (alodahl) [inline databases/relational-databases.md]: How do you model

File: databases/relational-databases.md [hunk 15]
```diff
@@ -117,12 +166,25 @@ Schema design:
...
-### Check for Understanding
-
+Solve the quizes mentioned below
 - [W3Schools quiz](http://www.w3schools.com/sql/sql_quiz.asp)
 - [JavaTPoint sql quiz](https://www.javatpoint.com/sql-quiz)
 - [TutorialsPoint quiz](https://www.tutorialspoint.com/sql/sql_online_quiz.htm)
 
+
+### Check for Understanding
+
```
Comment (alodahl) [inline databases/relational-databases.md]: Thanks for those questions! I just have some grammar comments.

File: databases/relational-databases.md [hunk 16]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+  - Front end people should have concepts and write very basic SQL
+  - Back end people spend a large amount of time with databases
+- Structure of a relational database; tables with columns and references to other tables
+- Few basic column types
+  - `varchar`
+  - `text`
+  - `integer`
+  - `primary key`, `auto increment`
+    - exact type varies depending on database type
+- General SQL statement pattern (verb-subject-optional modifiers)
```
Comment (alodahl) [inline databases/relational-databases.md]: Needs an explanation

File: databases/relational-databases.md [hunk 17]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+
+- The specific problems a database solves
+  - Persistence of information
+  - Centralized information
+  - Search stored information
+  - Form relationships between sets of information
+- Users of databases
+  - Front end people should have concepts and write very basic SQL
+  - Back end people spend a large amount of time with databases
+- Structure of a relational database; tables with columns and references to other tables
```
Comment (alodahl) [inline databases/relational-databases.md]: Needs an explanation Or reference

File: databases/relational-databases.md [hunk 18]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+  - `varchar`
+  - `text`
+  - `integer`
+  - `primary key`, `auto increment`
+    - exact type varies depending on database type
+- General SQL statement pattern (verb-subject-optional modifiers)
+- The four basic SQL verbs and their use
+  - `SELECT`
+    - without modifiers
+    - selecting specific columns: \*, specific columns, AS column aliasing
```
Comment (alodahl) [inline databases/relational-databases.md]: Needs an explanation Or example

File: databases/relational-databases.md [hunk 19]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+- Few basic column types
+  - `varchar`
+  - `text`
+  - `integer`
+  - `primary key`, `auto increment`
+    - exact type varies depending on database type
+- General SQL statement pattern (verb-subject-optional modifiers)
+- The four basic SQL verbs and their use
+  - `SELECT`
+    - without modifiers
```
Comment (alodahl) [inline databases/relational-databases.md]: Needs an explanation Or example

File: databases/relational-databases.md [hunk 20]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+  - Back end people spend a large amount of time with databases
+- Structure of a relational database; tables with columns and references to other tables
+- Few basic column types
+  - `varchar`
+  - `text`
+  - `integer`
+  - `primary key`, `auto increment`
+    - exact type varies depending on database type
+- General SQL statement pattern (verb-subject-optional modifiers)
+- The four basic SQL verbs and their use
```
Comment (alodahl) [inline databases/relational-databases.md]: Needs an explanation Or example

File: databases/relational-databases.md [hunk 21]
```diff
@@ -55,10 +55,59 @@ To understand, query, and insert information into a relational database, a techn
...
+    - from another table with constraints `(INSERT INTO a SELECT x, y, z FROM b WHERE x = 'a thing')`
+    - Would be good to demonstrate type mismatch (`INSERT` '1' into an integer column), but if using MySQL
+      to teach this doesn't work as MySQL automatically type casts depending on server config
+  - `UPDATE...SET`
+    - without modifiers (entire table)
+    - with modifier constraints (same as `SELECT`)
+  - `DELETE FROM`
+    - without modifiers (entire table)
+    - with modifier constraints (same as `SELECT`)
+- DDL to create tables
```
Comment (alodahl) [inline databases/relational-databases.md]: What’s a ddl?

Comment (yati1998) [reply databases/relational-databases.md]: data Definition language. or in short we can mean language or command to create tablr

**Comment by yati1998** [comment]
@alodahl I have done the required changes. Please do review.

File: databases/relational-databases.md (hunk 21 above)
Comment (alodahl) [reply databases/relational-databases.md]: Can you describe that here or provide a link please

File: databases/relational-databases.md (hunk 3 above)
Comment (alodahl) [reply databases/relational-databases.md]: 👆🏼

File: databases/relational-databases.md (hunk 15 above)
Comment (alodahl) [reply databases/relational-databases.md]: Fixes look good.  Does the above outline provide answers to all these questions? Please look at each one.  If not, the info needs to be added.

File: databases/relational-databases.md (hunk 20 above)
Comment (alodahl) [reply databases/relational-databases.md]: 👆🏼

File: databases/relational-databases.md [hunk 22]
```diff
@@ -31,6 +32,36 @@ To understand, query, and insert information into a relational database, a techn
 
 ### Specific Things To Teach
 
+- The specific problems a database solves
```
Comment (alodahl) [inline databases/relational-databases.md]: Please make sure all of this info is provided later on in the outline. If it isn’t, please add it.

File: databases/relational-databases.md [hunk 23]
```diff
@@ -31,6 +32,36 @@ To understand, query, and insert information into a relational database, a techn
...
+- The four basic SQL verbs and their use
+- Bonus topic for the advanced: foreign key constraints
+- Explain different databases with slightly different dialects (MySQL - PostgreSQL)
+- Explain existence of other types of non-relational, non-sql databases that will be encountered
+
+### Materials
+
+#### Education
+
+- [Khan Academy's introductory SQL, videos](https://www.khanacademy.org/computing/computer-programming/sql/sql-basics/v/welcome-to-sql)
```
Comment (alodahl) [inline databases/relational-databases.md]: Are apprentices told how to use this resource in lesson or one of the practices? If not, please provide that.

File: databases/relational-databases.md (hunk 21 above)
Comment (alodahl) [reply databases/relational-databases.md]: 🌠

File: databases/relational-databases.md [hunk 24]
```diff
@@ -71,42 +77,67 @@ To understand, query, and insert information into a relational database, a techn
   - `integer`
   - `primary key`, `auto increment`
     - exact type varies depending on database type
-- General SQL statement pattern (verb-subject-optional modifiers)
 - The four basic SQL verbs and their use
   - `SELECT`
-    - without modifiers
-    - selecting specific columns: \*, specific columns, AS column aliasing
+    - The SELECT statement is used to select data from a database.
```
Comment (alodahl) [inline databases/relational-databases.md]: This is great! Be sure to use code notation for any code in this file.

File: databases/relational-databases.md (hunk 23 above)
Comment (yati1998) [reply databases/relational-databases.md]: Yes I have mentioned that in lesson

**Comment by yati1998** [comment]
@alodahl please do review my latest PR.

**Comment by yati1998** [comment]
@alodahl do I need to make further changes to this or am I done with this issue? I made all the addressed changes.

**Comment by alodahl** [comment]
I don't think your title accurately describes your changes.

**Comment by yati1998** [comment]
@alodahl yes I have checked both check for understanding and  specific things to teach, all the points are mentioned in the lesson

**Comment by yati1998** [comment]
> I don't think your title accurately describes your changes or the issue requirements. Please fix.

Sorry for the inconvenience I will take care of that.

**Comment by yati1998** [comment]
> I don't think your title accurately describes your changes or the issue requirements. Please fix.

Sorry for the inconvenience I will take care of that. 

> I don't think your title accurately describes your changes or the issue requirements. Please fix.

Do you want me to change the PR name or any commit?

**Comment by yati1998** [comment]
@alodahl I have made the changes required.

**Comment by alodahl** [comment]
> > I don't think your title accurately describes your changes or the issue requirements. Please fix.
> 
> Sorry for the inconvenience I will take care of that.
> 
> > I don't think your title accurately describes your changes or the issue requirements. Please fix.
> 
> Do you want me to change the PR name or any commit?

Your PR Title.

File: databases/relational-databases.md [hunk 25]
```diff
@@ -39,69 +69,128 @@ To understand, query, and insert information into a relational database, a techn
 - Users of databases
   - Front end people should have concepts and write very basic SQL
   - Back end people spend a large amount of time with databases
-- Structure of a relational database; tables with columns and references to other tables
+- [Structure of a relational database](https://www.lucidchart.com/pages/database-diagram/database-design)
```
Comment (alodahl) [inline databases/relational-databases.md]: Yay!  Now please add this link and the 4 basic SQL verbs - and all links used in the lesson, for that matter - to the ### Materials section.

File: databases/relational-databases.md [hunk 26]
```diff
@@ -31,6 +32,35 @@ To understand, query, and insert information into a relational database, a techn
...
+- Bonus topic for the advanced: foreign key constraints
+- Explain different databases with slightly different dialects (MySQL - PostgreSQL)
+- Explain existence of other types of non-relational, non-sql databases that will be encountered
+
+### Materials
+
+#### Education
+
+- [Khan Academy's introductory SQL, videos](https://www.khanacademy.org/computing/computer-programming/sql/sql-basics/v/welcome-to-sql)
+- [TutorialsPoint, detailed](https://www.tutorialspoint.com/sql/)
```
Comment (alodahl) [inline databases/relational-databases.md]: Where is this in the outline? If its not there, can you add it under ### Lesson?  Same with the link in line 50.

**Comment by yati1998** [comment]
@alodahl I have done the required changes please do review.

**Comment by yati1998** [comment]
@alodahl is the PR title correct now?

**Comment by alodahl** [comment]
Yes, it would be even better if the title mentions what youre changing!  Like:
sql.md: fix things to teach, add supplemental section & questions in check for understanding

Another tip: the convention for git commit messages is to write them in *command tense*, like you are telling the commit to make certain changes. If it starts to get long, try to summarize it more but add details to the description.

**Comment by yati1998** [comment]
@alodahl I will take care of that next time onwards.  Thanks for the suggestions.  I will keep this in mind

File: databases/relational-databases.md [hunk 27]
```diff
@@ -39,69 +76,132 @@ To understand, query, and insert information into a relational database, a techn
...
+     - Speed. True relational databases support client server architecture. Populate times for large ODBC databases will not grow as quickly as populate times for Paradox databases.
+     - Structured Query Language (SQL), which is the basis for relational database management systems. SQL is the standard data access language of the American National Standards Institute (ANSI) and the International Standards Organization (ISO). If you are familiar with SQL, you can access any RDBMS.
+     - Incremental data storage, which gives you a historical perspective of the data.
+     - Client/Server configuration, which lets you run Visualizer on one machine while storing data on another machine.
+     - A single interface, which provides integrated data that can be shared across platforms, networks, and devices.
+- Describing an ORM:
+  - Object-relational mapping (ORM) is a programming technique in which a metadata descriptor is used to connect object code to a relational database. Object code is written in object-oriented programming (OOP) languages such as Java or C#. ORM converts data between type systems that are unable to coexist within relational databases and OOP languages.
```
Comment (alodahl) [inline databases/relational-databases.md]: https://www.techopedia.com/definition/24200/object-relational-mapping--orm

File: databases/relational-databases.md [hunk 28]
```diff
@@ -39,69 +76,132 @@ To understand, query, and insert information into a relational database, a techn
...
+  - Below is an example of its usage:
+    ```sql
+      CREATE TABLE Orders (
+      OrderID int NOT NULL,
+      OrderNumber int NOT NULL,
+      PersonID int,
+      PRIMARY KEY (OrderID),
+      FOREIGN KEY (PersonID) REFERENCES Persons(PersonID);
+    ```
+The table containing the foreign key is called the child table, and the table containing the candidate key is called the referenced or parent table.
```
Comment (alodahl) [inline databases/relational-databases.md]: https://en.wikipedia.org/wiki/Foreign_key

Comment (yati1998) [reply databases/relational-databases.md]: do I need to add this link?

**Comment by yati1998** [comment]
@alodahl I really sorry for this misconception. I really didn't have an idea that we can't put anything from sources. Since it was informative stuff I chose to do so. I will keep that from next time.

**Comment by alodahl** [comment]
You are welcome to use outside sources, but you must give proper credit, otherwise it is pretending these are your own words.  I’m going to need you to go back to any PRs you’ve made with something like this and cite the source.

**Comment by yati1998** [comment]
> You are welcome to use outside sources, but you must give proper credit, otherwise it is pretending these are your own words. I’m going to need you to go back to any PRs you’ve made with something like this and cite the source. Please search through all of them yourself. This is pretty serious, and I don’t want to have a curriculum full of plagiarism.
> 
> I can’t give you any more issues until you’ve done this. @manaswinidas @Kundan28 @praveenscience @saberprashant

Okay sure. I will do that all.

**Comment by yati1998** [comment]
@alodahl do I need to provide the link as well or just mention the credits?

**Comment by yati1998** [comment]
> You are welcome to use outside sources, but you must give proper credit, otherwise it is pretending these are your own words. I’m going to need you to go back to any PRs you’ve made with something like this and cite the source. Please search through all of them yourself. This is pretty serious, and I don’t want to have a curriculum full of plagiarism.
> 
> I can’t give you any more issues until you’ve done this. @manaswinidas @Kundan28 @praveenscience @saberprashant

@alodahl I have gone through all the previous Pr's and added credits wherever required. Can you now review my previous issue and mention the changes.

**Comment by Kundan28** [comment]
> @alodahl do I need to provide the link as well or just mention the credits?

You should provide the link too as just by name we are not giving proper credit to the owner.

**Comment by yati1998** [comment]
@alodahl please do review my commit.

File: databases/relational-databases.md [hunk 29]
```diff
@@ -39,69 +76,132 @@ To understand, query, and insert information into a relational database, a techn
...
   - `CREATE TABLE`
-- Bonus topic for the advanced: foreign key constraints
-- Explain different databases with slightly different dialects (MySQL - PostgreSQL)
-- Explain existence of other types of non-relational, non-sql databases that will be encountered
-  - examples key-value (redis), graph (Neo4j), streaming (cassandra), document (mongo)
 
-### Materials
+- Adavntages of Relational Database:
+  - Relational databases are critical to moving, sharing and maintaining the integrity of data in the modern workplace where users working with PCs, workstations, and mainframes all need access to the same data. Using an RDBMS offers the following advantages:
+     - Speed. True relational databases support client server architecture. Populate times for large ODBC databases will not grow as quickly as populate times for Paradox databases.
```
[the remark this answers is not in the document]
Comment (alodahl) [reply databases/relational-databases.md]: Advantages

File: databases/relational-databases.md (hunk 28 above)
Comment (alodahl) [reply databases/relational-databases.md]: YES

File: databases/relational-databases.md [hunk 30]
```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
...
   - `DELETE FROM`
     - without modifiers (entire table)
+      ```sql
+      DELETE FROM table_name;
+      ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+      DELETE FROM table_name WHERE condition;
+      ```
+- DDL or Data Definition Language actually consists of the SQL commands that can be used to define the database schema. It simply deals with descriptions of the database schema and is used to create and modify the structure of database objects in database.
```
Comment (alodahl) [inline databases/relational-databases.md]: https://www.geeksforgeeks.org/sql-ddl-dml-dcl-tcl-commands/
You got this from here

File: databases/relational-databases.md [hunk 31]
```diff
@@ -161,12 +262,39 @@ Schema design:
...
+### Supplemental Resources
+
+- [Medium: Omar El Gabry's Tutorials on Databases, Modeling, Design Process, & Normalization](https://medium.com/omarelgabrys-blog/database-introduction-part-1-4844fada1fb0)
+- [TutorialsPoint, detailed](https://www.tutorialspoint.com/sql/)
+- [Codecademy Tutorial, interactive](https://www.codecademy.com/learn/learn-sql)
+- [Learn SQL the hard way, textbook](https://learncodethehardway.org/sql/)
+- [GalaXQL, interactive, graphical](http://sol.gfxile.net/galaxql.html)
+- [PostgreSQL SQL, technical reference manual](https://www.postgresql.org/docs/current/static/sql.html)
+- [AgileData: Data Modeling 101](http://www.agiledata.org/essays/dataModeling101.html)
+- [LucidChart: What is a Database Model?](https://www.lucidchart.com/pages/database-diagram/database-models)
```
Comment (alodahl) [inline databases/relational-databases.md]: Double check these links. its not supplemental if you are assigning it in the outline.

File: databases/relational-databases.md [hunk 32]
```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
...
+      PRIMARY KEY (OrderID),
+      FOREIGN KEY (PersonID) REFERENCES Persons(PersonID);
+    ```
+The table containing the foreign key is called the child table, and the table containing the candidate key is called the referenced or parent table.
+- Other types of non-relational, non-sql databases that will be encountered
+  - key-value (redis)
+    - Redis is a key-value database (also known as a key-value store) that uses a simple key/value method to store data.
+       Strings are the simplest data type in Redis and are simple key/value entries.
+  - graph (Neo4j)
+    - Neo4j is a graph database management system developed by Neo4j, Inc. Described by its developers as an ACID-compliant transactional database with native graph storage and processing.
```
Comment (alodahl) [inline databases/relational-databases.md]: https://github.com/JNOSQL/artemis-demo/tree/master/artemis-demo-java-se/graph-neo4j
☹️

File: databases/relational-databases.md [hunk 33]
```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
...
+      OrderNumber int NOT NULL,
+      PersonID int,
+      PRIMARY KEY (OrderID),
+      FOREIGN KEY (PersonID) REFERENCES Persons(PersonID);
+    ```
+The table containing the foreign key is called the child table, and the table containing the candidate key is called the referenced or parent table.
+- Other types of non-relational, non-sql databases that will be encountered
+  - key-value (redis)
+    - Redis is a key-value database (also known as a key-value store) that uses a simple key/value method to store data.
+       Strings are the simplest data type in Redis and are simple key/value entries.
```
Comment (alodahl) [inline databases/relational-databases.md]: https://www.shellhacks.com/en/redis-set-get-key-value-redis-cli/

File: databases/relational-databases.md [hunk 34]
```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
...
+  - graph (Neo4j)
+    - Neo4j is a graph database management system developed by Neo4j, Inc. Described by its developers as an ACID-compliant transactional database with native graph storage and processing. 
+  - document (mongo)
+    - MongoDB is a document database: each record in a MongoDB collection is document. Documents are a structure composed of file and value pairs,         similar to JSON objects or other mapping data types.
```
Comment (alodahl) [inline databases/relational-databases.md]: https://docs.mongodb.com/getting-started/cpp/documents/

File: databases/relational-databases.md (hunk 32 above)
Comment (yati1998) [reply databases/relational-databases.md]: > I'm going to give you one more chance to double-check any files you have worked on under GSSoC and cite them.

I am really sorry for such a mistake. While checking the previous commits, I just slipped off from my mind to check the present commit.

File: databases/relational-databases.md [hunk 35]
```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
...
+        VALUES (value1, value2, value3, ...);
+          ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+       INSERT INTO table_name (column1, column2, column3, ...)
+        VALUES (value1, value2, value3, ...);
+        ```
     - from another table `(INSERT INTO a SELECT x, y, z from b)`
     - from another table with constraints `(INSERT INTO a SELECT x, y, z FROM b WHERE x = 'a thing')`
     - Would be good to demonstrate type mismatch (`INSERT` '1' into an integer column), but if using MySQL
```
Comment (alodahl) [inline databases/relational-databases.md]: can you rephrase this?  This outline will be read directly by the apprentices

File: databases/relational-databases.md [hunk 36]
```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
...
+        ```
+
   - `DELETE FROM`
     - without modifiers (entire table)
+      ```sql
+      DELETE FROM table_name;
+      ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+      DELETE FROM table_name WHERE condition;
```
Comment (alodahl) [inline databases/relational-databases.md]: Also, did you get line 89-156 from somewhere in particular? If so, cite it.

**Comment by yati1998** [comment]
@alodahl  I have made the required changes. Please do review them.

File: databases/relational-databases.md (hunk 30 above)
Comment (alodahl) [reply databases/relational-databases.md]: 👆

File: databases/relational-databases.md [hunk 37]
```diff
@@ -174,8 +175,8 @@ Examples of DDL commands:
...
      - Client/Server configuration, which lets you run Visualizer on one machine while storing data on another machine.
      - A single interface, which provides integrated data that can be shared across platforms, networks, and devices.
-- [Describing an ORM](https://www.techopedia.com/definition/24200/object-relational-mapping--orm):
-  - Object-relational mapping (ORM) is a programming technique in which a metadata descriptor is used to connect object code to a relational database. Object code is written in object-oriented programming (OOP) languages such as Java or C#. ORM converts data between type systems that are unable to coexist within relational databases and OOP languages.
+- Describing an ORM:
+  - Object-relational mapping (ORM) is a programming technique in which a metadata descriptor is used to connect object code to a relational database. Object code is written in object-oriented programming (OOP) languages such as Java or C#. ORM converts data between type systems that are unable to coexist within relational databases and OOP languages. [From Technopedia](https://www.techopedia.com/definition/24200/object-relational-mapping--orm)
```
Comment (alodahl) [inline databases/relational-databases.md]: 🎯

File: databases/relational-databases.md [hunk 38]
```diff
@@ -155,15 +155,16 @@ Also briefly review: [LucidChart: What is a Database Model?](https://www.lucidch
       ```sql
       DELETE FROM table_name WHERE condition;
       ```
-- [DDL](https://www.geeksforgeeks.org/sql-ddl-dml-dcl-tcl-commands/) or Data Definition Language actually consists of the SQL commands that can be used to define the database schema. It simply deals with descriptions of the database schema and is used to create and modify the structure of database objects in database.
+- The above syntax are from [w3schools.](https://www.w3schools.com/sql/sql_select.asp)
```
Comment (alodahl) [inline databases/relational-databases.md]: these are great, but please indent so it is part of the previous section.

File: databases/relational-databases.md [hunk 39]
```diff
@@ -155,15 +155,16 @@ Also briefly review: [LucidChart: What is a Database Model?](https://www.lucidch
...
+- DDL or Data Definition Language actually consists of the SQL commands that can be used to define the database schema. It simply deals with descriptions of the database schema and is used to create and modify the structure of database objects in database.
 Examples of DDL commands:
   - CREATE – is used to create the database or its objects (like table, index, function, views, store procedure and triggers).
   - DROP – is used to delete objects from the database.
   - ALTER-is used to alter the structure of the database.
   - TRUNCATE–is used to remove all records from a table, including all spaces allocated for the records are removed.
   - COMMENT –is used to add comments to the data dictionary.
   - RENAME –is used to rename an object existing in the database.
-
+- The above definition is from [GeeksforGreeks](https://www.geeksforgeeks.org/sql-ddl-dml-dcl-tcl-commands/)
```
Comment (alodahl) [inline databases/relational-databases.md]: these are great, but please indent so it is part of the previous section.

**Comment by yati1998** [comment]
@alodahl  I have made the required changes.

File: databases/relational-databases.md (hunk 5 above)
Comment (alodahl) [inline databases/relational-databases.md]: This should be under the title, not part of the title

File: databases/relational-databases.md (hunk 35 above)
Comment (yati1998) [reply databases/relational-databases.md]: @alodahl are you talking about this change? I have removed this line as I didn't feel it important to be here. I am sorry to say but I couldn't find out which change are you talking about. I will be grateful to you if you mention them.

Comment (alodahl) [reply databases/relational-databases.md]: - Would be good to demonstrate type mismatch (`INSERT` '1' into an integer column), but if using MySQL to teach this doesn't work as MySQL automatically type casts depending on server config

Comment (yati1998) [reply databases/relational-databases.md]: Shall I rephase this? Can I remove this from outline?

**Comment by yati1998** [comment]
@alodahl am I done with this issue?

File: databases/relational-databases.md [hunk 40]
```diff
@@ -31,6 +33,43 @@ To understand, query, and insert information into a relational database, a techn
...
+
+- [SQL Fiddle](http://sqlfiddle.com)
+
+### Lesson
+
+- [Slides](https://drive.google.com/open?id=1xK7_t_yJcu4RcBkj0Gv-t5uyBCNr0g4cHKqAJSxNwY0) | [Video Walkthrough of Slides](https://drive.google.com/file/d/1V0bk3fH_8PsRE3Vz4J3qe3TTiqBClT6y/view)
+
+Also briefly review: [LucidChart: What is a Database Model?](https://www.lucidchart.com/pages/database-diagram/database-models). Be sure to read the [Relational Model](https://www.lucidchart.com/pages/database-diagram/database-models?a=1) section, and briefly skim over the rest just so you're aware that there are many types of database models. The only one we're working with here is the relational model, so don't worry about learning the rest yet. 
+
+- Go through the complete tutorial mentioned in the Materials Section. [SQL Fiddle](http://sqlfiddle.com) is the tool where you can implement your sql knowledge. Practically implement creating a database schema and perform operations on it using the tool.
```
Comment (alodahl) [inline databases/relational-databases.md]: Which tutorial do they need to complete? Is sql fiddle a tutorial

Comment (yati1998) [reply databases/relational-databases.md]: No its a tool. I have mentioned in the description.

Comment (alodahl) [reply databases/relational-databases.md]: Then which tutorial do they need to complete?

Comment (yati1998) [reply databases/relational-databases.md]: The one that are mentioned in material section

Comment (alodahl) [reply databases/relational-databases.md]: None of them say tutorial. Please just tell me which one, or change the sentence yourself.

Comment (alodahl) [reply databases/relational-databases.md]: Are we having translation problems here?

Comment (alodahl) [reply databases/relational-databases.md]: Do you mean” go through every material in the Materials section” ?

Comment (yati1998) [reply databases/relational-databases.md]: Yes.

Comment (alodahl) [reply databases/relational-databases.md]: I see, thank you.
